02_oop/GL_TASK_2/mainwindow.cpp

```diff
@@ -0,0 +1,301 @@
+#include "mainwindow.h"
+#include "ui_mainwindow.h"
+#include <QLabel>
+
+// for add file to table
+void addFile(QString fileName, QTableWidget *table);
+
+// for export edit files
+void saveToFile(QTableWidget *table, int count);
+
+MainWindow::MainWindow(QWidget *parent)
+    : QMainWindow(parent)
+    , ui(new Ui::MainWindow)
+{
+    ui->setupUi(this);
+
+    setWindowFlags(Qt::Dialog | Qt::MSWindowsFixedSizeDialogHint);
+    // save theme -------------
+    /*
+    QDir dir("/GL_TASK_2/Sources");
+    //QString s;
+
+    //s = dir.relativeFilePath("/home/mary/file.txt");
+    QString filN = dir.relativeFilePath("../Other files/settingsSave.txt");
+    QFile  nFile(filN);
+
+    if (!nFile.open(QFile::ReadWrite  | QFile::Text))
+    {
+      qDebug() << "could not open the file";
+      return;
+    }
+
+    QTextStream str(&nFile);
+    qDebug() << str.readLine();
+
+    nFile.close();
+    */
+
+    on_actiondef_triggered();
+}
+
+MainWindow::~MainWindow()
+{
+    delete ui;
+}
+
+// ADD FILES --- --- --- --- --- ---
+void MainWindow::on_add_file_1_clicked()
+{
+    QString fileName = QFileDialog::getOpenFileName(this, "Open A File", "C:\\" );
+    QTableWidget *table = ui->tableWidget_1;
+    addFile(fileName, table);
+
+}
+
+void MainWindow::on_add_file_2_clicked()
+{
+    QString fileName = QFileDialog::getOpenFileName(this, "Open A File", "C:\\" );
+    QTableWidget *table2 = ui->tableWidget_2;
+    addFile(fileName, table2);
+
+}
+
+void addFile(QString fileName, QTableWidget *table){
+
+    if (fileName != "")
+    {
+        QFile file(QFileInfo(fileName).absoluteFilePath());
```

Comment:
You can replace it by
```
if (fileName.isEmpty()) {
    throw std::runtime_error{ "some error message" };
}
```
it will allow you to remove one level of nesting